map.go

```diff
@@ -344,11 +354,11 @@ func splitHash(h uint64) (h1, h2) {
 	return h1((h & h1Mask) >> 7), h2(h & h2Mask)
 }
 
-func probeStart(hi h1, groups int) uint32 {
-	return fastModN(uint32(hi), uint32(groups))
+func probeStart[S constraints.Unsigned](hi h1, groups int) S {
+	return fastModN(S(hi), S(groups))
 }
 
 // lemire.me/blog/2016/06/27/a-fast-alternative-to-the-modulo-reduction/
-func fastModN(x, n uint32) uint32 {
-	return uint32((uint64(x) * uint64(n)) >> 32)
+func fastModN[S constraints.Unsigned](x, n S) S {
+	return S((uint64(x) * uint64(n)) >> 32)
```

Comment:
This is the part that will not work with `uint64`: it will always return values in 32-bit range, so those buckets will be unused.

Reply:
I see. Would returning `unsafe.Sizeof(S)` help here? Or is this a bit-op trick that can not be rescued?

Reply:
I don't think it can be rescued without having to do the _usual_ modulo division. See the link above that explains how the trick works.

Reply:
This [writeup](https://github.com/sipa/writeups/tree/main/uniform-range-extraction#c-version) shows a possible version to support uint64 ranges.

Perhaps there's a method here for assigning the correct `fastmodN` for the map instantiation. It, however, wouldn't be inlined by the compiler.

Reply:
Thank you for the link! It's always great to learn something new.

Maybe it would make sense to just have a fork of this repo (swiss64?) with the 64-bit version? It should be quite easy to keep it up to date (I guess there shouldn't be many or frequent incoming changes), the API would be clean (`swiss.Map` vs `swiss64.Map`) and the code would be easier and faster to follow. WDYT?

Reply:
I'm not sure. That seems like a dolthub call.

Reply:
the fast modulo with 64 bits support can be as simple as 
```go
func fastModN(x, n uint) uint {
    h, _ := bits.Mul(x, n)
    return h
}
```
**but** this requires the high bits of x to be set (which is not the case for H1). So just call it with `x=H1<<7`